rattr/analyser/util.py

```diff
@@ -810,6 +810,81 @@ def walrus_in_rhs(node: AnyAssign) -> bool:
         return False
 
 
+def namedtuple_in_rhs(node: AnyAssign) -> bool:
+    """Return `True` if the RHS contains a namedtuple creation."""
+    _iterable = (ast.Tuple, ast.List)
+
+    def _target_is_namedtuple(call: ast.Call) -> bool:
+        # HACK
```

Comment:
As mentioned earlier, the best you can do is track the full path in the current scope from imports

Reply:
The only reason I do not force `typing.NamedTuple` is to allow `internal.NamedTuple` for some `class NamedTuple(typing.NamedTuple): ...`. I agree that it would be nice to go ruff style and track imports :)